src/main/java/groupM/players/mcts/TreeNode.java

```diff
@@ -97,6 +104,29 @@ void mctsSearch() {
         }
     }
 
+    abstract Comparator<TreeNode> getPruningComparator();
+
+    List<AbstractAction> unprunedActions() {
+        return children.keySet().stream().filter(a->(children.get(a) == null || !children.get(a).pruned)).collect(toList());
+    }
+
+    List<TreeNode> unprunedNodes() {
+        return children.values().stream().filter(a->!a.pruned).collect(toList());
+    }
+
+    private void pruneChildren() {
+        long childLog = Math.round(this.player.params.pruneAlpha*Math.log(children.size()));
+        long retainedChildren =
+                Math.max(childLog,this.player.params.minRetained);
+        List<TreeNode> childNodes = unprunedNodes();
+        childNodes.sort(getPruningComparator());
+        Iterator<TreeNode> childIt = childNodes.iterator();
+        while(unprunedActions().size() > retainedChildren) {
+            TreeNode node = childIt.next();
+            node.pruned = node.nVisits>=20;
```

Comment:
Shouldn't this be a parameter?

Reply:
it could be yes, did it like this as it was just a constant in the paper rather than variable. I will update it